hypervisor/arch/x86/pagetable.c

```diff
@@ -269,6 +315,63 @@ static void modify_or_del_pdpte(const uint64_t *pml4e, uint64_t vaddr_start, uin
  * type: MR_DEL
  * delete [vaddr_base, vaddr_base + size ) memory region page table mapping.
  */
+/**
+ * @brief Modify or delete the mappings associated with the specified address range.
+ *
+ * This function modifies the properties of an existing mapping or deletes it entirely from the page table. The input
+ * address range is specified by [vaddr_base, vaddr_base + size). It is used when changing the access permissions of a
+ * memory region or when freeing a previously mapped region. This operation is critical for dynamic memory management,
+ * allowing the system to adapt to changes in memory usage patterns or to reclaim resources.
+ *
+ * Normal case behaviors will successfully modify the properties of a mapping or remove it. Error case behaviors include
```

Comment:
A few comments here for the normal case behavior and error case behavior:
1. `invalid parameters` is vague here. In general, we need to specify what kind of conditions are considered to be invalid.
2. A few functionalities implemented in this interface is not covered, let's add them here.

For normal case behaviors:
 - Large page is split to small pages under specific conditions. We can say, under xxx conditions, a large page table is split into next level page table. We can eliminate the details about how to do the split, but the condition is important. We will also need to mention that the execution right is recovered by the callback function when 2MB page is split to 4KB pages, because this behavior is special and may not be easily derived from the `split` word.
 - When mapping is deleted, corresponding pages will be sanitized.

For error case behaviors:
 - The handling of a non-existing PML4E and PDPTE/PDE/PTE is different, let's explicitly state them.

Reply:
Got it. Thanks!